src/legacy/core_plugins/ui_metric/public/services/telemetry_analytics.ts

```diff
@@ -0,0 +1,56 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import { createReporter, Reporter, UiStatsMetricType } from '@kbn/analytics';
+
+let telemetryReporter: Reporter;
+
+export const setTelemetryReporter = (aTelemetryReporter: Reporter): void => {
+  telemetryReporter = aTelemetryReporter;
+};
+
+export const getTelemetryReporter = () => {
+  return telemetryReporter;
+};
+
+export const getUiStatsReporter = (appName: string) => (
```

Comment:
Minor nit, but I would find `createUiStatsReporter` to be a little clearer from the consumer's. When I see `get` I expect to be retrieving something that has been `set`. But with `create` I know I am creating something for my localized use case.

Reply:
++